Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp

```diff
@@ -449,23 +448,39 @@ void WebResourceLoadStatisticsStore::setLoginStatus(RegistrableDomain&& domain,
 {
     ASSERT(RunLoop::isMain());
 
-    if (loggedInStatus == IsLoggedIn::LoggedIn) {
-        auto loginStatusToSet = lastAuthentication && lastAuthentication->hasExpired() ? std::nullopt : std::optional(WTFMove(lastAuthentication));
-        if (loginStatusToSet)
-            loginStatusToSet->setTimeToLive(WebCore::LoginStatus::TimeToLiveLong);
-        auto pair = std::make_pair(loggedInStatus, loginStatusToSet);
-        m_loginStatus.set(domain, WTFMove(pair));
-    } else
-        m_loginStatus.remove(domain);
-    completionHandler();
+    auto loginStatusToSet = lastAuthentication && lastAuthentication->hasExpired() ? std::nullopt : std::optional(WTFMove(lastAuthentication));
+    if (loginStatusToSet)
+        loginStatusToSet->setTimeToLive(WebCore::LoginStatus::TimeToLiveLong);
+
+    postTask([this, domain = WTFMove(domain).isolatedCopy(), loggedInStatus, loginStatusToSet = WTFMove(*loginStatusToSet).isolatedCopy(), completionHandler = WTFMove(completionHandler)]() mutable {
```

Comment:
... But here you dereference `loginStatusToSet` unconditionally, which would crash if `loginStatusToSet` was nullopt.